hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerImpl.java

```diff
@@ -417,12 +436,12 @@ public int minPipelineLimit(Pipeline pipeline) {
   @Override
   public void activatePipeline(PipelineID pipelineID)
       throws IOException {
-    lock.lock();
+    acquireWriteLock();
     try {
       stateManager.updatePipelineState(pipelineID.getProtobuf(),
               HddsProtos.PipelineState.PIPELINE_OPEN);
     } finally {
-      lock.unlock();
+      releaseWriteLock();
     }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 439:445, 457:463.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.